src/client/componentsMealsharing/ReservationSpecific.js

```diff
@@ -0,0 +1,174 @@
+import React, { useEffect, useState } from "react";
+import "./Meal.css";
+import { Link } from "react-router-dom";
+
+function ReservationSpecific({ id, max_reservations }) {
+  const [fetchReservations, setFetchReservations] = useState([]);
+  const [seat, setSeat] = useState("");
+  const [phone, setPhone] = useState("");
+  const [name, setName] = useState("");
+  const [mail, setMail] = useState("");
+  const [reservation, setReservation] = useState([]);
+  const [feedback, setFeedback] = useState("");
+  useEffect(() => {
+    fetchReservationResult();
+  }, []);
+
+  const fetchReservationResult = async () => {
+    const response = await fetch(`http://localhost:3000/api/reservations`);
+    const Data = await response.json();
+    setFetchReservations(Data);
+  };
+  // get the number of guests
+  const titles = fetchReservations
+    .filter((item) => item.meal_id == id)
+    .map((items) => Number(items.number_of_guests));
+  const totalGuests = titles.reduce((current, total) => current + total, 0);
+  const seatsLeft = max_reservations - totalGuests;
+
+  const addReservation = async (e) => {
+    e.preventDefault();
+    const newObj = {
+      number_of_guests: seat,
+      contact_phonenumber: phone,
+      contact_name: name,
+      contact_email: mail,
+      created_date: "2022-04-29",
```

Comment:
I would have the created_date be done on the backend instead of the frontend :) as the database can automatically set the created_date :)

Reply:
how to set the date automatically in the backend? Thanks for telling.